src/FpsCounter.cpp

```diff
@@ -14,44 +14,27 @@
 
 using namespace NAS2D;
 
-const unsigned int FPS_COUNTS_SIZE = 25;
-unsigned int FPS_COUNTS[FPS_COUNTS_SIZE] = { 0 };
-
-unsigned int	CURRENT_TICK		= 0;
-unsigned int	LAST_TICK			= 0;
-unsigned int	TICK_DELTA			= 0;
-unsigned int	INDEX				= 0;
-unsigned int	ACCUMULATOR			= 0;
-
-
-/**
- * FpsCounter c'tor
- */
-FpsCounter::FpsCounter()
-{}
-
 
 /**
  * Gets an average count of Frames per Second.
  */
 unsigned int FpsCounter::fps()
 {
-	LAST_TICK = CURRENT_TICK;
-	CURRENT_TICK = SDL_GetTicks();
-
-	TICK_DELTA = CURRENT_TICK - LAST_TICK;
+	auto lastTick = currentTick;
+	currentTick = SDL_GetTicks();
 
-	if (TICK_DELTA == 0) { TICK_DELTA = 1; }
+	auto tickDelta = currentTick - lastTick;
+	if (tickDelta == 0) { tickDelta = 1; }
 
-	FPS_COUNTS[++INDEX] = 1000 / TICK_DELTA;
+	fpsCounts[++fpsCountIndex] = 1000 / tickDelta;
 
-	if (INDEX >= FPS_COUNTS_SIZE) { INDEX = 0; }
+	if (fpsCountIndex >= FpsCountsSize) { fpsCountIndex = 0; }
```

Comment:
This is a slower version of `fpsCountIndex %= FpsCountsSize` due to the branch.

Also, avoid starting a variable name with a capital letter.

Reply:
The `%` operator effectively uses division. Is is unlikely to be faster than a branch. An exception would be when the modulus is a power of 2, in which case it can be translated to a fast bit mask. In this case though, we are not using a power of 2.

The `%` operator would be less code, and I was tempted to use it.

Reply:
The `FpsCountsSize` is a constant, hence the UpperCamelCase.

Reply:
Offline testing shows a branch vs modulo to be between 3x and 500 microseconds faster depending on the cutoff and if it's a power of two or not. I'm okay with that.

---

As for the name being capitalized "because it's a constant" (I assume for "at a glance" as well), that's not a good reason. The type system and Intellisense does that for you.

Reply:
It's a pretty common convention to use UpperCamelCase for global constants. You can already find examples of it in the code base.

This is for defined constants. Not `const&` references to mutable variables.

Reply:
But it's not a global constant. It's a private static member variable.

Reply:
A `static` class variable is effectively the same as a global `const`. They're used in equivalent contexts. They're both stored in the same area of the data section. They're initialized the same way.

Actually, I might also recommend UpperCamelCase even if it was function local.

Reply:
The wiki doesn't have a style guide. :/ Until otherwise, I'll just live with it.

Reply:
I've sent Leeor a question about it. It should probably be his decision. Most of the existing global constants used `UPPER_SNAKE_CASE`, which should be reserved for macros, and which we are moving away from. There doesn't seem to be well established practice for what to move to.